codegen/post_gen_hooks.go

```diff
@@ -330,3 +341,162 @@ func writeAndFormat(path string, data []byte) error {
 	}
 	return FormatGoFile(path)
 }
+
+// WorkflowMockGenHook returns a PostGenHook to generate endpoint workflow mocks
+func WorkflowMockGenHook(h *PackageHelper, t *Template) PostGenHook {
+	return func(instances map[string][]*ModuleInstance) error {
+		errChanSize := 0
+		shouldGenMap := map[*ModuleInstance][]*EndpointSpec{}
+		for _, instance := range instances["endpoint"] {
```

Comment:
It seems that this code only generates the initializer for `endpoint` type. What about user defined types like `task`?

Reply:
User defined types would require corresponding user defined hooks to generate the initializers.